pkg/kubectl/cmd/util/factory_object_mapping.go

```diff
@@ -73,33 +73,15 @@ func NewObjectMappingFactory(clientAccessFactory ClientAccessFactory) ObjectMapp
 	return f
 }
 
+// objectLoader attempts to perform discovery against the server, and will fall back to
+// the built in mapper if necessary. It supports unstructured objects either way, since
+// the underlying Scheme supports Unstructured. The mapper will return converters that can
+// convert versioned types to unstructured and back.
 func (f *ring1Factory) objectLoader() (meta.RESTMapper, runtime.ObjectTyper, error) {
-	mapper := legacyscheme.Registry.RESTMapper()
 	discoveryClient, err := f.clientAccessFactory.DiscoveryClient()
 	if err != nil {
-		glog.V(5).Infof("Object loader was unable to retrieve a discovery client: %v", err)
-		return mapper, legacyscheme.Scheme, nil
-	}
-	mapper = meta.FirstHitRESTMapper{
-		MultiRESTMapper: meta.MultiRESTMapper{
-			discovery.NewDeferredDiscoveryRESTMapper(discoveryClient, legacyscheme.Registry.InterfacesFor),
-			legacyscheme.Registry.RESTMapper(), // hardcoded fall back
-		},
-	}
-
-	// wrap with shortcuts, they require a discoveryClient
-	mapper = NewShortcutExpander(mapper, discoveryClient)
-	return mapper, legacyscheme.Scheme, nil
-}
-
-// TODO: unstructured and structured discovery should both gracefully degrade in the presence
-// of errors, and it should be possible to get access to the unstructured object typers no matter
-// whether the server responds or not. Make the legacy scheme recognize object typer, then we can
-// safely fall back to the built in restmapper as a last resort.
-func (f *ring1Factory) unstructuredObjectLoader() (meta.RESTMapper, runtime.ObjectTyper, error) {
-	discoveryClient, err := f.clientAccessFactory.DiscoveryClient()
-	if err != nil {
-		return nil, nil, err
+		glog.V(3).Infof("Unable to get a discovery client to find server resources, falling back to hardcoded types: %v", err)
```

Comment:
Why should we continue doing this?

Reply:
Fallback for local.  You still need a RESTMapper if you can't contact the server.  If discovery doesn't work, all but a few commands want to fail partially.